CHANGELOG.md

```diff
@@ -1,5 +1,62 @@
+* All current AKS API versions silently ignore unknown fields. An unknown field is a field that isn't part of the AKS API. AKS API version 2024-01-01, 2024-01-02-preview and all subsequent API versions will change this behavior. Unknown fields in a request will result in the request being rejected with an error stating that the unknown field is not understood. This change only impacts new API versions and won't impact you unless you update to use an API version 2024-01-01 or later. Existing API calls (via Azure Resource Manager templates or otherwise) will continue to function as-is.
+ 
+### Release notes
+
+* Features
+  * [Planned Maintenance](https://learn.microsoft.com/azure/aks/planned-maintenance) and [node-image Upgrade channel](https://learn.microsoft.com/azure/aks/auto-upgrade-node-os-image) is available in Azure Portal. 
+  * [Associate capacity reservation groups to node pools](https://learn.microsoft.com/azure/aks/manage-node-pools#associate-capacity-reservation-groups-to-node-pools-preview) is now generally available. 
+
+* Preview features
+  * AKS 1.29 is in preview. 
+  * [Control Plane Metrics (API server, ETCD, Cluster Autoscaler, etc)](https://learn.microsoft.com/azure/aks/monitor-control-plane-metrics) are now available in Managed Prometheus for Public preview. 
+ 
+
+* Bug Fixes 
+  * Fix the issue that the new CRD "azureapplicationgatewayrewrites.appgw.ingress.azure.io " is not delivered into the overlay cluster. This problem occurs during cluster upgrades.
+  * Enable [HonorPVReclaimPolicy](https://kubernetes.io/blog/2021/12/15/kubernetes-1-23-prevent-persistentvolume-leaks-when-deleting-out-of-order/#pv-reclaim-policy-with-kubernetes-v1-23) for CSI drivers on AKS 1.27+. 
+  * Fix preflight validation failure when BYOK is required in vmss node pool setup by Azure Policy.
+  * Node Auto Provision can now be enabled when aadProfiles, including ServerAppID, ClientAppID, ServerAppSecret, are being set.
+  * Istio-based service mesh add-on's istiod and ingress images updated to 1.18.7-hotfix.20240210 and 1.19.7 for asm-1-18 and asm-1-19 respectively. User needs to restart the workload pods to trigger re-injection of the newer patch version of istio-proxy. Vulnerabilities CVE-2024-23322, CVE-2024-23323, CVE-2024-23324, CVE-2024-23325, and CVE-2024-23327 have been addressed in these patch versions. More information can be found [here](https://learn.microsoft.com/azure/aks/istio-upgrade).
+  * Fix typo in KEDA helm chart from keda-admission-webhooks to keda-admission. 
+  * Fix a bug that the id field is missing in get/list privateEndpointConnections.
+  * ⁠For upgrades using Azure API version prior to 2022-04-01 API, Kubernetes version might fall behind current Kubernetes version which could block GETs/PUTs. This can be solved by doing a PUT with the current Kubernetes version but a fix is being worked on by AKS. 
+
+* Behavioral Change
+  * Update the Agentpool Profile protocol to include the new PodIPAllocationMode property.
```

Comment:
if it's a new property, what's the change in existing behavior for users? can we add more details on that part

Reply:
With 1.29 version we are introducing couple of component breaking changes per Pull Request #266245 · MicrosoftDocs/azure-docs-pr (github.com)](https://github.com/MicrosoftDocs/azure-docs-pr/pull/266245)

Tigera-Operator 1.30.7 up from 1.28.13 and azurefile-csi-driver 1.29.3 up from 1.29.2.

Reply:
breaking change lists new versions for tigera operator and csi drivers in the linked table. can we describe the behavior change?

Reply:
Drop CSI as its a patch change. On tigera there quite a few changes so lets link to their release notes - https://github.com/tigera/operator/releases/tag/v1.30.8